src/elements/content-sidebar/DocGenSidebar/DocGenSidebar.tsx

```diff
@@ -0,0 +1,161 @@
+    const [sidebarState, setSidebarState] = React.useState<State>({
+        hasError: false,
+        loading: false,
+        tags: {
+            text: [],
+            image: [],
+        },
+        jsonPaths: {
+            textTree: {},
+            imageTree: {},
+        },
+    });
+    const tagsToJsonPaths = (tags: DocGenTag[]): JsonPathsMap => {
+        const jsonPathsMap: JsonPathsMap = {};
+        tags.forEach(tag => {
+            tag.json_paths.forEach(jsonPath => {
+                const paths = jsonPath.split('.');
+                let currentObject: JsonPathsMap | {} = jsonPathsMap;
+                paths.forEach(segment => {
+                    if (isJsonPathsMap(currentObject)) {
+                        if (!(segment in currentObject)) {
+                            (currentObject as JsonPathsMap)[segment] = {};
+                        }
+                        currentObject = (currentObject as JsonPathsMap)[segment];
+                    }
+                });
+            });
+        });
+
+        return jsonPathsMap;
+    };
+
+    const loadTags = () => {
+        if (getDocGenTags) {
+            setSidebarState({ ...sidebarState, loading: true });
+            getDocGenTags()
+                .then((response: DocGenTemplateTagsResponse) => {
+                    if (response) {
+                        // anything that is not an image tag for this view is treated as a text tag
+                        const textTags = response?.data?.filter(tag => tag.tag_type !== 'image') || [];
+                        const imageTags = response?.data?.filter(tag => tag.tag_type === 'image') || [];
+                        setSidebarState({
+                            ...sidebarState,
+                            tags: {
+                                text: textTags,
+                                image: imageTags,
+                            },
+                            loading: false,
+                            jsonPaths: {
+                                textTree: tagsToJsonPaths(textTags),
+                                imageTree: tagsToJsonPaths(imageTags),
+                            },
+                        });
+                    } else {
+                        setSidebarState({ ...sidebarState, loading: false });
+                    }
+                })
+                .catch(() => setSidebarState({ ...sidebarState, loading: false, hasError: true }));
+        }
+    };
+
+    React.useEffect(() => {
+        loadTags();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const { hasError, tags, loading } = sidebarState;
+
+    const hasNoTags = tags.image.length + tags.text.length === 0;
+
+    return (
+        <SidebarContent sidebarView={SIDEBAR_VIEW_METADATA} title={intl.formatMessage(messages.docgenTags)}>
+            <div className={classNames('docgen-sidebar', { center: hasNoTags || hasError || loading })}>
```

Comment:
no because that breaks the underline in the header. Also its child with class 'bcs-scroll-content-wrapper' has position absolute which breaks styling for the padding.